### ISSUE TITLE
require is defined, causing an error, due to weird hacky browser/node compatibility

### ISSUE THREAD

**Issue by NullVoxPopuli** [rwjblue/ember-cli-cjs-transform#26]
This line: https://github.com/dchest/tweetnacl-js/blob/master/nacl-fast.js#L2364

from ember-cli-build.js
```
  app.import('node_modules/tweetnacl/nacl-fast.js', {
    using: [
      { transformation: 'cjs', as: 'tweetnacl'}
    ]
  });
```

Error:
```
Error while processing route: login Could not find module `crypto` imported from `tweetnacl` Error: Could not find module `crypto` imported from `tweetnacl`
```

This is how I'm importing it:
```
import * as NaCl from 'tweetnacl';
console.log(NaCl);
```

**Comment by bekzod** [comment]
@NullVoxPopuli I think what is happening here, is `tweetnacl` is using native node js dependency `crypto` which is not possible to use in browser-land

**Comment by acorncom** [comment]
@NullVoxPopuli are you hitting errors on this line? https://github.com/dchest/tweetnacl-js/blob/master/nacl-fast.js#L2366 If I'm reading that right, if you get to that line, then the browser you are using isn't properly offering the `crypto` module and you are then hitting snags because you're shifting into Node-land where the `require` option doesn't exist ...

**Comment by les2** [comment]
@NullVoxPopuli We have a working like this in our addon so that the transform works in apps:

```
included() {
    this._super.included.apply(this, arguments);
    this.import('node_modules/semver/semver.js', {
      using: [{ transformation: 'cjs', as: 'semver' }]
    });
  },

  // pass transforms from the addon to the apps (transformation for semver module)
  // Need this until a fix lands in ember-cli.
  importTransforms() {
    let transforms = this.eachAddonInvoke('importTransforms');
    return Object.assign({}, ...transforms);
  },
```

Not sure if your issue is the same one. YMMV.

**Comment by NullVoxPopuli** [comment]
> @NullVoxPopuli are you hitting errors on this line? https://github.com/dchest/tweetnacl-js/blob/master/nacl-fast.js#L2366 If I'm reading that right, if you get to that line, then the browser you are using isn't properly offering the crypto module and you are then hitting snags because you're shifting into Node-land where the require option doesn't exist ...

@acorncom yeah, that's the line I'm hitting.  I'm using chrome, so.. not sure if that'd have crypto or not. but I thought require was strictly node-only? idk.

> Not sure if your issue is the same one. YMMV.

@les2 unsure, I'm not familiar with importTransforms, and what it offers. if it maybe forces the load to be in a browser env, etc?

**Comment by les2** [comment]
try:
```
app.import('node_modules/tweetnacl/nacl-fast.js')
```

??

**Comment by NullVoxPopuli** [comment]
I have this atm:
```js
  app.import('node_modules/tweetnacl/nacl-fast.min.js');
  app.import('vendor/shims/tweetnacl.js');


(function() {
  function vendorModule() {
    'use strict';

    return {
      'default': self['nacl-fast.min'],
      __esModule: true,
    };
  }

  define('tweetnacl', [], vendorModule);
})();
```

![image](https://user-images.githubusercontent.com/199018/39444596-54f2e548-4c86-11e8-9cec-50b609673c1d.png)

**Comment by NullVoxPopuli** [comment]
ok, looks like I finally got it, thanks to @luxferresum on the ember slack

I learned a bit more about vendored assets.

the `self` var in the shim is like the `window` and I needed to look at how the vendored module was defining a global variable to know what to put in the square brackets.

so, my shim should be `self['nacl']`, because that's what nacl-fast(.min).js is defining.

I thought it was just based off the file name.
